examples_tests/31.SkinningDataBenchmark/main.cpp

```diff
@@ -4,396 +4,385 @@
+        meshBuffer->setIndexCount(geometryObject.indexCount);
+        meshBuffer->setIndexType(geometryObject.indexType);
+        meshBuffer->setIndexBufferBinding(std::move(geometryObject.indexBuffer));
+        
+        auto newInputParams = geometryObject.inputParams;
+        newInputParams.enabledAttribFlags |= 0b10000u;
+        newInputParams.enabledBindingFlags |= 0b10u;
+        newInputParams.bindings[1].inputRate = EVIR_PER_VERTEX;
+        newInputParams.bindings[1].stride = 0u;
+        newInputParams.attributes[4].binding = 1u;
+        newInputParams.attributes[4].format = EF_R16_UINT;
+        newInputParams.attributes[4].relativeOffset = 0u;
+
+        SBufferBinding<ICPUBuffer> boneIDBuffer;
+        boneIDBuffer.offset = 0u;
+        boneIDBuffer.buffer = core::make_smart_refctd_ptr<ICPUBuffer>(meshBuffer->calcVertexCount() * sizeof(uint16_t));
+
+        uint16_t* buffPtr = static_cast<uint16_t*>(boneIDBuffer.buffer->getPointer());
+        for (int i = 0; i < meshBuffer->calcVertexCount(); i++)
+            *(buffPtr + i) = boneID;
+
+
+        meshBuffer->setVertexBufferBinding(std::move(boneIDBuffer), 1);
+
+        auto pipeline = core::make_smart_refctd_ptr<ICPURenderpassIndependentPipeline>(nullptr, nullptr, nullptr, newInputParams, blendParams, geometryObject.assemblyParams, rasterParams);
+        meshBuffer->setPipeline(std::move(pipeline));
+    };
+
+    const core::vector4du32_SIMD diskBlockDim(5u, 5u, 5u);
+    const size_t diskCount = diskBlockDim.x * diskBlockDim.y * diskBlockDim.z;
+
+    std::vector<uint16_t> tesselation(diskCount);
+
+    //get random tesselation for disks
+    {
+        std::random_device rd;
+        std::mt19937 mt(rd());
+        std::uniform_int_distribution<uint32_t> dist(15, 160 - 1);
 
-        memcpy(_dst + boneSize*i, mat3x4, sizeof(mat3x4));
-        memcpy(_dst + boneSize*i + sizeof(mat3x4)/sizeof(float), mat3, sizeof(mat3));
+        //TODO: test
+        //`dist(mt) | 0x0001` so vertexCount is always odd (only when diskCount is odd as well)
+        std::generate(tesselation.begin(), tesselation.end(), [&]() { return dist(mt) | 0x0001; });
     }
 
-    const size_t perInstance = boneSize * _boneCnt;
-
-    for (size_t i = 1u; i < _instCnt; ++i)
-        memcpy(_dst + i*perInstance, _dst, sizeof(float)*perInstance);
+    core::vector<uint16_t> indices(16000);
+    std::iota(indices.begin(), indices.end(), 0u);
 
-    return sizeof(float) * perInstance * _instCnt;
-}
-size_t convertBuf3(const float* _src, float* _dst, const size_t _boneCnt, const size_t _instCnt)
-{
-    const size_t size_mat3_dst = 12u; // 12 floats
-    const size_t size_mat4x3_src = 12u; // 12 floats
-    const size_t size_mat4x3_dst = 16u; // 16 floats
-    const size_t offset_mat3_dst = _boneCnt * size_mat4x3_dst;
-    for (size_t i = 0u; i < _boneCnt; ++i)
+    core::vector<ICPUMeshBuffer*> disks(diskCount);
+    std::generate(disks.begin(), disks.end(), []() { return _IRR_NEW(ICPUMeshBuffer); });
```

Comment:
use `smart_refctd_ptr`s instead

Reply:
instead of `new` do `core::make_smart_refctd_ptr<Type>(constructor args...)`